src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -0,0 +1,410 @@
+        $servers = '/' !== $baseUrl && '' !== $baseUrl ? [new Model\Server($baseUrl)] : [];
+        $paths = new Model\Paths();
+        $links = [];
+        $schemas = [];
+
+        foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
+            $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+            $resourceShortName = $resourceMetadata->getShortName();
+
+            // Items needs to be parsed first to be able to reference the lines from the collection operation
+            list($itemOperationLinks, $itemOperationSchemas) = $this->collectPaths($resourceMetadata, $resourceClass, OperationType::ITEM, $context, $paths, $links, $schemas);
+            $links = array_merge($links, $itemOperationLinks);
+            $schemas += $itemOperationSchemas;
+            list($collectionOperationLinks, $collectionOperationSchemas) = $this->collectPaths($resourceMetadata, $resourceClass, OperationType::COLLECTION, $context, $paths, $links, $schemas);
+            $links = array_merge($links, $collectionOperationLinks);
+            $schemas += $collectionOperationSchemas;
+        }
+
+        $securitySchemes = $this->getSecuritySchemes();
+        $securityRequirements = [];
+
+        foreach (array_keys($securitySchemes) as $key) {
+            $securityRequirements[$key] = [];
+        }
+
+        return new OpenApi($info, $servers, $paths, new Model\Components(new \ArrayObject($schemas), new \ArrayObject(), new \ArrayObject(), new \ArrayObject(), new \ArrayObject(), new \ArrayObject(), new \ArrayObject($securitySchemes)), $securityRequirements);
+    }
+
+    private function collectPaths(ResourceMetadata $resourceMetadata, string $resourceClass, string $operationType, array $context, Model\Paths $paths, array $links, array $schemas = []): array
+    {
+        $resourceShortName = $resourceMetadata->getShortName();
+        if (null === $operations = OperationType::COLLECTION === $operationType ? $resourceMetadata->getCollectionOperations() : $resourceMetadata->getItemOperations()) {
+            return [$links, $schemas];
+        }
+
+        foreach ($operations as $operationName => $operation) {
+            $path = $this->getPath($resourceShortName, $operationName, $operation, $operationType);
+            $method = $resourceMetadata->getTypedOperationAttribute($operationType, $operationName, 'method', 'GET');
+            list($requestMimeTypes, $responseMimeTypes) = $this->getMimeTypes($resourceClass, $operationName, $operationType, $resourceMetadata);
+            $operationId = $operation['openapi_context']['operationId'] ?? lcfirst($operationName).ucfirst($resourceShortName).ucfirst($operationType);
+            $linkedOperationId = 'get'.ucfirst($resourceShortName).ucfirst(OperationType::ITEM);
+            $pathItem = $paths->getPath($path) ?: new Model\PathItem();
+
+            /** @TODO support multiple formats for schemas? */
```

Comment:
It is already supported in the current version IIRC.

Reply:
it's buggy af and not really supported everywhere in the current version :p